### ISSUE TITLE
Gira eNet lights not setting accurate brightness level from Homey

### ISSUE THREAD

**Issue by JOR200** [mnordseth/enet-homeassistant#13]
Thanks for a very good integration for Gira eNet Server in HA. 
I primary use HA as a gateway between eNet Server and Homey and have a small isue as follows: 
The brightness level for eNet lights are not set accurate in HA when controlled from Homey. This vary a bit, but seem to be 0%, 1% or 2% off in the whole range of the scale.
Settng 1% in Homey will set to 0% (Off) in HA. HA and eNet server seem to agree on the level.
This may be a HACS issue. All my other lights (wich are not HACS integrations) are have matching levels in HA and Homey. 
Only my eNet lights brighness level differs slightly in HA and Homey.

**Comment by mnordseth** [comment]
Hi, I am not able to test this use case as I do not use or have Homey.

I do not think it is related HACS as that is purely a way of installing integrations. The enet integration implement the Light interface in Home Assistant, and should behave similar to Philips Hue and other integrations. 

Can you provide some debug output from the enet plugin with desciption of what you do from homey and expected results?

**Comment by JOR200** [comment]
Thanks for responding to this issue.
I just added IKEA Trådfri bridge integration (not HACS) to HA and the brightness levels matches exactly in Homey, HA and IKEA mobile app. I'm not sure how to get the debug output from enet plugin but here's som info from HA:
![image](https://user-images.githubusercontent.com/59866495/228204671-9ba42049-346a-42a8-b5d2-63013cd5bcb6.png)

Levels in HA and eNet server matches exactly, but they do not match exactly in Homey. 
Here's some value examples for an eNet light:

Homey              HA / eNet server
100                    100
99                      97
80                      80
70                      69
60                      60
50                      49
15                      14
1                        0

The one thing I can think of is the way values are converted between a scale of 0-100% and 0-255. Is this done differently with HACS integrations than ordinary integrations?

The screenshot shows 100% brightness values for light.benk_vaskerom (IKEA) as 254 and 100% brightness value for light.hjornelampe (eNet) as 255.

**Comment by JOR200** [comment]
My table for brighness level did not show correctly... 
First column is for Homey, the second is for HA/eNet server.

**Comment by JOR200** [comment]
Hi. Yes, I agree. The presicion is not very important except for the differnce between 0% and 1%. My eNet lights are set to turn on with 1% brightness in the eNet server. They are emidiatly turned of again because Homey believes they are set to 0% (Off).
I only see this behaviour for my eNet lights. All other lights does not seem to have this precision issue. That's why I wondered if this was an issue with HACS. How can I set the lights to 1%? (1/100)255=2.55? If I set the lights to 2% in Homey, HA and eNet server says they are at 1% and they stay on. This is not the case for other lights wich don't use HACS. 

Just trying to figure out if this is a bug and where it can be addressed. And why only an issue with the eNet lights? I do not have other lights with HACS integration.

**Comment by JOR200** [comment]
Hi, again. 
I really think this is a presicion issue too, and I think I know how to fix it quite easily.

When brightness level is written to the eNet server, a fixed value of 0.49% should be added to the value you already have. This will round the values upwards to nearest percentage instead of downwards except for a byte value of zero. 0 will give 0.49% (0,0049) on the eNet server witch will be rounded downwards to 0. The same goes for a byte value of FF (255) witch will give a value of 100.49% on the eNet server. This value should be rounded downwards to 100%. Now, when you read back the values from eNet server, you should pretty much get a match with the value that was originally written. I am pretty sure this will solve my isuue since a byte value of 0 will be the only value that will set eNet server to 0%. 

Do you agree with me?

**Comment by JOR200** [comment]
Why only add 0.49 to the write value and not 0.5? Because 0.5 will also round the persentage value of zero up to 1% and the persentage value of 100 will be rounded upwards to 101. You do not wan't that.

**Comment by JOR200** [comment]
This should do it. I'm pretty sure this will also handle the exact values for 0% and 100% correctly. Looking forward to test the new verison. Thanks a lot...

**Comment by JOR200** [comment]
Hi. I've testet the new version. When controlled from Homey, everything matches up. Values in Homey, HA and eNet server matches perfectly. I now can set the brightness to any given level. But, there is a but... Looks like HA does not respond very well anymore to the levels when set in eNet mobile app. Also eNet server values seem to be set slightly off when set directly in HA. Do you think similar modifications are needed when reading values from eNet server?

**Comment by mnordseth** [comment]
I am unable to reproduce that. For me the values match up between the enet
app and home assistant (both directions).

**Comment by JOR200** [comment]
I'm glad you did not reproduce the behavior. I was affraid this fix had broken functionality for other users. Strange thing. Never the less, I did a another reatart of HA and tested again. Now all values match up in all directions. eNet server, Homey and HA show the same. I don't know why the reading of eNet server values did not work properly after installing the update and restart HA, but now it does.

I will do some more testing, but expect this to work perfectly.

Thanks a lot...

**Comment by JOR200** [comment]
Everything seems to work as desired. I'm closing the issue. Thanks again.

**Comment by JOR200** [comment]
Hi. I have reopened this issue as I still have an issue.
I have set the turn on brightness in eNet server to 1%.
When lights are turned on from eNet mobile app or wall switch the lights are set to 1% brughness as expected. A second or so after that, HA turns the lights off again. 

I have the same behaviour with or without Homey.

Maybe the value read from eNet server schould be rounded upwards in the same way as for writing?

**Comment by mnordseth** [comment]
Hi, I am unable to reproduce this. In my setup the values set from the enet mobile app are reflected perfectly in Home Assistant.

You say HA turn the lights off. Is that just the representation in HA or does the actual light turn off. If the latter is the case, how do you know HA is controlling it? What is the logbook output when you do this?

**Comment by JOR200** [comment]
Hi. This is what Logbook says: 
![image](https://user-images.githubusercontent.com/59866495/232680900-075289d3-e4f1-4491-9cd0-e6ada99df4a9.png)

I took it for granted that HA did this, but when I turn off my HA server I still have this behaviour. It must be a bug in eNet server itselves. 

Here's how I configured most of my lights in eNet server:
![image](https://user-images.githubusercontent.com/59866495/232681956-b129cc53-9273-4c62-bfa7-245c193a03f8.png)
![image](https://user-images.githubusercontent.com/59866495/232682118-89b8a45a-9874-4b1e-8370-9ec9755bbf9e.png)

Can you try to set up like this and just turn the lights on with a wall switch? 
Do you see the same behaviour?

I have eNet server version 2.4.0.

**Comment by JOR200** [comment]
By setting Switch-on brightness to 1% or 2%, the lights will just turn off again. 3% and above will keep the lights on.

**Comment by nestorix1343** [comment]
Just my two cents: When just looking at my Jung eNet setup, some lights will not light up if the minimum settings is below 5%. Some other LED lights light up from 3%. Some LED lights light up randomly when turning on at such a low percentage, so I set the minimum based on a test what the minimum is.

It appears to be dependent on the make and brand of the LED lights.

I tried several settings like the phase settings but that did not seem to help.
